src/shared/Api.js

```diff
@@ -1,34 +1,36 @@
-import { useEffect, useState } from "react";
+import { useEffect, useState, useMemo } from "react";
 
-function useMovies(genre, getRandom) {
+function useMovies(genre, enableRandomization) {
   const [movies, setMovies] = useState([]);
 
-  const fetchMoviesData = () => {
-    let apiUrl = "https://itunes.apple.com/us/rss/topmovies/limit=80/json";
-    if (genre) {
-      apiUrl =
-        "https://itunes.apple.com/us/rss/topmovies/limit=80/genre=${genre}/json";
-    }
-
-    fetch(apiUrl)
-      .then((response) => {
-        return response.json();
-      })
-      .then((data) => {
-        setMovies(data.feed.entry);
-      });
-  };
+  const fetchMoviesData = async () => {
+    const apiUrl = genre
+      ? `https://itunes.apple.com/us/rss/topmovies/limit=80/genre=${genre}/json`
```

Comment:
W dwóch wierszach masz to samo, wyciągnij do consta
https://itunes.apple.com/us/rss/topmovies/limit=80/